### ISSUE TITLE
Package elixir-ls

### ISSUE THREAD

**Issue by Br1ght0ne** [NixOS/nixpkgs#73231]
**Project description**
From [project's homepage](https://github.com/elixir-lsp/elixir-ls/):
> The Elixir Language Server provides a server that runs in the background, providing IDEs, editors, and other tools with information about Elixir Mix projects. It adheres to the Language Server Protocol, a standard for frontend-independent IDE support. Debugger integration is accomplished through the similar VS Code Debug Protocol.

**Metadata**

* homepage URL: https://github.com/elixir-lsp/elixir-ls/
* source URL: https://github.com/elixir-lsp/elixir-ls/
* license: asl20
* platforms: unix

**Comment by Parasrah** [comment]
This is something I'd really like to see as well

**Comment by Parasrah** [comment]
Figured I'd take a crack at this myself, others can view it [here](https://github.com/NixOS/nixpkgs/compare/master...Parasrah:elixir-ls)

I didn't bother trying to build it from source because there is no way that I'm aware of to fetch hex dependencies for elixir projects right now that works reliably. I renamed the files to `elixir_ls_server.sh`, `elixir_ls_debugger.sh` and `elixir_ls_launch.sh`, and fixed the files to reference the correct paths given the name changes.

This does seem to have caused issues however, as the language server crashes after being launched and initialized by a language client. I think a potential solution is to create our own script that calls into those sh scripts so they don't have to live in `$out/bin` and therefore don't need to be renamed. That said I don't know when I'll get another chance to look into this so anyone is welcome to make their own attempt in the meantime.
